### PR TITLE
fix IRGraph init & Add regualizer

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?

1. fix init primitives in IR graph, all primitives should be initialized together, otherwise there will be result error.

2. Add regualizer for linear and convolution dnn layer

## How was this patch tested?

unit tests and training test

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/View.scala [hunk 1]
```diff
@@ -99,13 +98,18 @@ class View[T: ClassTag](val sizes: Array[Int])(
...
     } else {
       bse
     }
   }
 
+  private def batchSize(
+    input: Tensor[T], size: Array[Int], numberInputDims: Int, numElements: Int): Int = {
+    batch(input.nDimension(), input.size(), size, numberInputDims, numElements)
+  }
+
```
Comment (i8run) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/View.scala]: Use only 1 api is much better i think.

Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/View.scala]: ok, changed

**Comment by i8run** [comment]
LGTM.

**Comment by zhangxiaoli73** [comment]
jenkins pass:
http://mlt-bddl01.sh.intel.com:8080/view/PR/job/BigDL-PRVN-lite/358/

**Pull request merged**